src/librbd/image/RemoveRequest.cc

```diff
@@ -152,8 +152,10 @@ void RemoveRequest<I>::handle_trim_image(int r) {
   ldout(m_cct, 20) << "r=" << r << dendl;
 
   if (r < 0) {
-    lderr(m_cct) << "warning: failed to remove some object(s): "
+    lderr(m_cct) << "failed to remove some object(s): "
                  << cpp_strerror(r) << dendl;
+    send_close_image(r);
```

Comment:
I though it was rather a feature that we didn't abort if failed to remove some objects. I don't remember if I saw a real case when it could be useful though, anyway might this change deserve a separate PR (tracker ticket) to be more visible for users?

Reply:
I can't think of a good reason why we should just leave around data blocks. It was definitely something that was inherited from the original version of remove back when it was in "internal.cc". I can split it out to a separate PR, but I don't think anyone is going to look for it in PR descriptions.

Reply:
It was just a suggestion. As you wish.